### ISSUE TITLE
Polys get stuck when rebuilding if a resource is out

### ISSUE THREAD

**Issue by PalatinatePotatoFarmer** [Anuken/Mindustry#5087]
**Platform**: Android

**Build**: 126.2

**Issue**: 
The war story, if anyone wants to hear it ;-):
While playing the campaign, I got a bad whacking while defending sector 226 (which was the first "extreme" base I tackled so far... having only T3 units available, I was really struggling during the initial capture attempts (my core got destroyed at least twice... yes, I suck... ;-)) and I only made it by suicide bombing the core,  by repeatedly flying several Zenith loaded with 80 units of blast compound each into the enemy core).
So during that defense mission,
my defenses got maimed pretty badly during the last 5 waves - I wouldn't have made it if I had not had the deeply layered defenses set up during the original conquest of the sector, where I used sniping with long range turrets to destroy as much of the enemy base as possible. 
Since this sector is still threatened and since I dare not attack the two extinction level sectors next to it that threaten it yet (not until I have at least T4 units), I had hoped that building a load of polys to auto-reconstruct most parts of the base while I rebuild and upgrade the defenses would give me a quick way to get a chance of surviving the next attack. Alas, I am somewhat low on phase fabric (yes, I have launch pads on several sectors launching phase fabric and more to 226, but I am still not that good at mass producing it). I had initially built tons of RTGs, since Thorium is abundant on that map and since I have some trouble keeping impact reactors happy. So the Polys start rebuilding my hinterland, while I am away fixing those battered front lines. After a while, I go back to check progress - and find all polys stuck re-"building" an RTG, during which they ran out of phase fabric. And instead of moving to some other reconstruction project that would not require phase fabric, they just sit there, their build beam pointed on the RTG, without progress. >:-(

**The issue, in short form:**
Polys get stuck while auto-rebuilding when the block they are trying to rebuild requires a resource that is currently out. 
Expected behaviour: polys move on to some other reconstruction project nearby not requiring that particular resource. 
Nice to have: having a basic way to control the area where polys rebuild (without having to dive into writing logic ;-)).

**Link(s) to mod(s) used**: zero mods

**Save file**: coming soon... have to figure out where to find the campaign save game

**(Crash) logs**: no crash
---

*Place an X (no spaces) between the brackets to confirm that you have read the line below.*  
- [x] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [x] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by Anuken** [comment]
> Expected behaviour: polys move on to some other reconstruction project nearby not requiring that particular resource.

That's not a reasonable expectation, and this is not a bug. Items are frequently replenished continuously, and giving up on construction simply because all the items are not available immediately isn't always desirable behavior.
